### PR TITLE
Initialize model to nullptr

### PR DESCRIPTION
**Pull request by Montel**

### CODE REVIEW

File: examples/demo_threading/cpp/helpers/sensor.h [hunk 1]
```diff
@@ -49,6 +49,6 @@ private Q_SLOTS:
 
 private:
   ::std::optional<int> m_index;
-  EnergyUsageProxyModel* m_model;
+  EnergyUsageProxyModel* m_model = nullptr;
```
Comment (ahayzen-kdab) [inline examples/demo_threading/cpp/helpers/sensor.h:52]: Can you do this in the constructor implementation instead, to keep consistency with elsewhere.

```cpp
Sensor::Sensor(QObject* parent)
   : QObject(parent)
   , m_model(nullptr)
{
...
}
```

Comment (Montel) [reply examples/demo_threading/cpp/helpers/sensor.h:52]: Ok I will change it.

**Review by ahayzen-kdab — APPROVED** [review]
LGTM

**Comment by LeonMatthesKDAB** [comment]
This is why we need memory-safe languages :sweat_smile:

**Comment by ahayzen-kdab** [comment]
Even just smart pointers or sane initialise values :-)

**Pull request merged**